treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,141 @@
+            jp.testtype = jp.testtype.replace('e10s-devtools-chrome', 'devtools-chrome-e10s')
+            jp.testtype = jp.testtype.replace('gl-', 'webgl-')
+
+        key = jp.unique_identifier()
+        if key in ref_data_names:
+            # e.g. desktop-test-linux64-pgo/opt-reftest-13 or builder name
+            jobs.append(ref_data_names[key])
+        else:
+            logger.warning('We did not find job priority ({}) in the list of accepted jobs'.format(jp))
+
+    return jobs
+
+
+def _gecko_decision_task_request(project):
+    # In the case of a request from the Gecko decision task we don't care which priority is
+    # requested. We return either all jobs every 5th push or 90 minutes for that project or
+    # all TaskCluster high value jobs.
+    task_request = _update_task_request(project)
+    last_reset = task_request.seconds_since_last_reset()
+    job_priorities = []
+    for jp in _query_job_priorities(priority=0, excluded_build_system_type='buildbot'):
+        # We only add a job if it hasn't reached its timeout.
+        # If the timeout is zero it means that the job needs to always run
+        if last_reset < jp.timeout or jp.timeout == 0:
+            # Due to the priority of all high value jobs is 1, and we
+            # We need to return all jobs for every 5th push.
+            if task_request.counter % jp.priority != 0:
+                job_priorities.append(jp)
+
+    return _process(build_system='taskcluster', job_priorities=job_priorities)
+
+
+def _update_task_request(project):
+    '''Update a repository's counter and/or reset timer.'''
+    now = datetime.datetime.now()
+
+    try:
+        task_request = TaskRequest.objects.get(repository__name=project)
+        # Increase the counter of how many times the gecko decision task
+        # has contacted us
+        task_request.counter += 1
+        if task_request.has_expired():
+            task_request.last_reset = now
+        task_request.save()
+    except TaskRequest.DoesNotExist:
+        task_request = TaskRequest.objects.create(
+            repository=Repository.objects.get(name=project),
+            counter=1,
+            last_reset=now,
+            reset_delta=RESET_DELTA)
+
+    return task_request
+
+
+def _query_job_priorities(priority, excluded_build_system_type):
+    job_priorities = JobPriority.objects.all()
+    if priority != 0:
+        job_priorities = job_priorities.filter(priority=priority)
+    return job_priorities.exclude(buildsystem=excluded_build_system_type)
+
+
+def _validate_request(build_system_type, project):
+    if build_system_type not in ('buildbot', 'taskcluster', '*'):
+        raise SetaError('Valid build_system_type values are buildbot or taskcluster.')
+
+
+def seta_job_scheduling(project, build_system_type, priority=5, user_agent=None):
+    _validate_request(build_system_type, project)
+
+    if user_agent == GECKO_DECISION_TASK_USER_AGENT:
```

Comment:
Oh so there are actual writes occurring here?

If so, this shouldn't be a GET but by POST and there should be a Hawk API key used here, otherwise anyone can abuse it.

Either way, using User-Agent is a bit too magical, the behaviour difference should be differentiated by either GET vs POST or a url parameter.